### DISCUSSION TITLE
oh-my-posh found in the Microsoft Store is missing a valid latest package version

### DISCUSSION THREAD

**Discussion by CidDk** [JanDeDobbeleer/oh-my-posh#5576] [Troubleshoot]
The version of oh-my-posh found in the Microsoft Store is not compatible whit intunes Add App.
It simply states "The selected app does not have a valid latest package version."

**Comment by JanDeDobbeleer** [comment]
@CidDk I'm going to be honest, I have no idea what that means. Care to clarify?

**Reply by CidDk** [reply]
Sure @JanDeDobbeleer , as a Intune admin I like make the app available to the users in our organization.

There are many ways to do so, but I like to use the "Microsoft Store and Microsoft Intune integration" as it take only takes a few seconds. In Intune I navigate to "Add App > Microsoft Store App" use the search function here I finds oh-my-posh, and hit the bottom to make the magic work.

We have about 70 Microsoft Store Apps we make available this way and never before had a error.
I know i can simply make a script to install it using winget and upload it, (properly even use the Microsoft Store Id for winget).
But I would like it to be fixed for all of us.

I sadly never submitted any apps to the Microsoft Store so I don't know exactly how the store works in the back but i found this:
https://rozemuller.com/windows-store-app-not-supported-in-preview-in-intune/

> The problem with Windows Store applications is that not all the mandatory properties are filled in by the application developer.

btw I'm dyslexic and english is not my native language so I hope the absents of red lines means this is not total nonsense.

**Reply by CidDk** [reply]
looking up oh-my-posh in the windows store we find that "PackageVersion" is "Unknown".
using postman get > https://storeedgefd.dsx.mp.microsoft.com/v9.0/packageManifests/XP8K0HKJFRXGCK

{
    "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.PackageManifestResponse, StoreEdgeFD",
    "Data": {
        "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.PackageManifestData, StoreEdgeFD",
        "PackageIdentifier": "XP8K0HKJFRXGCK",
        "Versions": [
            {
                "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.PackageManifestVersion, StoreEdgeFD",
                "PackageVersion": "Unknown",
                "DefaultLocale": {
                    "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.DefaultLocale, StoreEdgeFD",
                    "PackageLocale": "en",
                    "Publisher": "jandedobbeleer",
                    "PublisherUrl": "https://ohmyposh.dev/",
                    "PrivacyUrl": "https://ohmyposh.dev/privacy",
                    "PublisherSupportUrl": "http://support@ohmyposh.dev/",
                    "PackageName": "oh-my-posh",
                    "License": "Copyright 2022 Jan De Dobbeleer\n\nPermission is hereby granted, free of charge, to any person obtaining a copy of this software and associated\ndocumentation files (the \"Software\"), to deal in the Software without restriction, including without limitation\nthe rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,\nand to permit persons to whom the Software is furnished to do so, subject to the following conditions:\n\nThe above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.\n\nTHE SOFTWARE IS PROVIDED \"AS IS\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE\nWARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR\nCOPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,\nARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.",
                    "Copyright": "2022 Jan De Dobbeleer",
                    "ShortDescription": "Oh My Posh is an executable that allows you to set a custom prompt on any shell. To get started after the installation, follow the guide over at https://ohmyposh.dev/docs/installation/windows#next to configure your prompt.",
                    "Description": "Oh My Posh is an executable that allows you to set a custom prompt on any shell. To get started after the installation, follow the guide over at https://ohmyposh.dev/docs/installation/windows#next to configure your prompt.",
                    "Tags": [
                        "oh-my-posh",
                        "oh my posh",
                        "powershell",
                        "terminal",
                        "prompt"
                    ],
                    "Agreements": [
                        {
                            "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.AgreementDetail, StoreEdgeFD",
                            "AgreementLabel": "Category",
                            "Agreement": "Productivity"
                        },
                        {
                            "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.AgreementDetail, StoreEdgeFD",
                            "AgreementLabel": "Pricing",
                            "Agreement": "Free"
                        },
                        {
                            "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.AgreementDetail, StoreEdgeFD",
                            "AgreementLabel": "Free Trial",
                            "Agreement": "No"
                        },
                        {
                            "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.AgreementDetail, StoreEdgeFD",
                            "AgreementLabel": "Terms of Transaction",
                            "AgreementUrl": "https://aka.ms/microsoft-store-terms-of-transaction"
                        },
                        {
                            "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.AgreementDetail, StoreEdgeFD",
                            "AgreementLabel": "Seizure Warning",
                            "AgreementUrl": "https://aka.ms/microsoft-store-seizure-warning"
                        },
                        {
                            "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.AgreementDetail, StoreEdgeFD",
                            "AgreementLabel": "Store License Terms",
                            "AgreementUrl": "https://aka.ms/microsoft-store-license"
                        }
                    ]
                },
                "Locales": [],
                "Installers": [
                    {
                        "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.SparkInstaller, StoreEdgeFD",
                        "InstallerSha256": "eb5af6793d264083a8810c3f42f035e8568d12654a377b4b98dacd6fa1d64a41",
                        "InstallerUrl": "https://github.com/JanDeDobbeleer/oh-my-posh/releases/download/v23.10.0/install-amd64.exe",
                        "InstallerLocale": "en",
                        "MinimumOSVersion": "0.0.0.0",
                        "InstallerSwitches": {
                            "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.InstallerSwitch, StoreEdgeFD",
                            "Silent": "/VERYSILENT /CURRENTUSER /INSTALLER=ws"
                        },
                        "InstallerSuccessCodes": [],
                        "ExpectedReturnCodes": [],
                        "Architecture": "x64",
                        "InstallerType": "exe",
                        "Markets": {
                            "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.Markets, StoreEdgeFD",
                            "AllowedMarkets": [
                                "US"
                            ]
                        },
                        "DownloadCommandProhibited": false
                    },
                    {
                        "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.SparkInstaller, StoreEdgeFD",
                        "InstallerSha256": "b1c7eec34dc86b6942daf7813c0cba1a7b0d5997bc00236215b6b150c5df074a",
                        "InstallerUrl": "https://github.com/JanDeDobbeleer/oh-my-posh/releases/download/v23.10.0/install-arm64.exe",
                        "InstallerLocale": "en",
                        "MinimumOSVersion": "0.0.0.0",
                        "InstallerSwitches": {
                            "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.InstallerSwitch, StoreEdgeFD",
                            "Silent": "/VERYSILENT /CURRENTUSER /INSTALLER=ws"
                        },
                        "InstallerSuccessCodes": [],
                        "ExpectedReturnCodes": [],
                        "Architecture": "arm64",
                        "InstallerType": "exe",
                        "Markets": {
                            "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.Markets, StoreEdgeFD",
                            "AllowedMarkets": [
                                "US"
                            ]
                        },
                        "DownloadCommandProhibited": false
                    },
                    {
                        "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.SparkInstaller, StoreEdgeFD",
                        "InstallerSha256": "33c221bb70c4e7335c71000b833f89a6ea65b244188ba91df522f75f07af83f7",
                        "InstallerUrl": "https://github.com/JanDeDobbeleer/oh-my-posh/releases/download/v23.10.0/install-386.exe",
                        "InstallerLocale": "en",
                        "MinimumOSVersion": "0.0.0.0",
                        "InstallerSwitches": {
                            "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.InstallerSwitch, StoreEdgeFD",
                            "Silent": "/VERYSILENT /CURRENTUSER /INSTALLER=ws"
                        },
                        "InstallerSuccessCodes": [],
                        "ExpectedReturnCodes": [],
                        "Architecture": "x86",
                        "InstallerType": "exe",
                        "Markets": {
                            "$type": "Microsoft.Marketplace.Storefront.StoreEdgeFD.BusinessLogic.Response.PackageManifest.Markets, StoreEdgeFD",
                            "AllowedMarkets": [
                                "US"
                            ]
                        },
                        "DownloadCommandProhibited": false
                    }
                ]
            }
        ]
    }
}

**Reply by JanDeDobbeleer** [reply]
@CidDk I can't solve this. The version is extracted in the provided package URL and the [submission API](https://learn.microsoft.com/en-us/windows/apps/publish/store-submission-api) doesn't contain a field to explicitly update that. You'll have to have a chat with the MS team here.
